### PR TITLE
Forums

### PR DESCRIPTION
**Pull request by Tsche**
`++done` in forums
bump discord.py and aiohttp
remove auto channels
parse new rule format

### CODE REVIEW

File: src/cogs/forums.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from src import config as conf
+from src.util import permissions
+
+
+class Forums(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.mapping = {}
+
+        for channel_id, close_tag in conf.closeable.items():
```
Comment (Yuhanun) [inline src/cogs/forums.py:13]: Order of these two elements isn’t guaranteed tho…

File: src/cogs/forums.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            if close_tag is None:
+                continue
+
+            channel = self.bot.get_channel(channel_id)
+            if not isinstance(channel, discord.ForumChannel):
+                continue
```
Comment (Yuhanun) [inline src/cogs/forums.py:19]: Message maybe?

File: src/cogs/forums.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @commands.hybrid_command(aliases=['close'], with_app_command=True)
+    async def done(self, ctx: commands.Context):
+        if not isinstance(ctx.channel, discord.Thread):
+            return
+
+        if ctx.channel.parent.id not in conf.closeable:
+            return
+
+        if ctx.channel.id == ctx.message.id:
+            await ctx.send("No.")
```
Comment (Yuhanun) [inline src/cogs/forums.py]: What’s this condition and can we make the error more explicit?

also; “return await ctx.send(…)”

**Review by Yuhanun — CHANGES_REQUESTED** [review]

File: src/cogs/forums.py (hunk 2 above)
Comment (Tsche) [reply src/cogs/forums.py:19]: nah, that's just to check we can actually set tags. No point in looking for a tag if there cannot be any

File: src/cogs/forums.py (hunk 1 above)
Comment (Tsche) [reply src/cogs/forums.py:13]: `.items()` always yields a tuple of (key, value). If you fuck up config that's on you, we do not validate config at all anyway.

Comment (Yuhanun) [reply src/cogs/forums.py:13]: you're right I misread this as "I expect 2 items in the array", like I saw `channel_id, close_tag = x.items()`, reviewedc on phone sorry

**Pull request merged**